sklearn/inspection/_h_statistic.py

```diff
@@ -0,0 +1,260 @@
+"""Friedman and Popescu's H-Statistic"""
+
+import itertools
+
+import numpy as np
+
+from ..base import is_classifier, is_regressor
+from ..utils import Bunch, _get_column_indices, _safe_assign, _safe_indexing
+from ..utils.random import sample_without_replacement
+from ..utils.validation import _check_sample_weight, check_is_fitted
+
+
+def _calculate_pd_brute_fast(estimator, X, feature_indices, grid, sample_weight=None):
+    """Fast version of _calculate_partial_dependence_brute()
+
+    Returns np.array of size (n_grid, ) or (n_ngrid, output_dimension).
+    """
+
+    if is_regressor(estimator):
+        pred_fun = estimator.predict
+    elif is_classifier(estimator) and hasattr(estimator, "predict_proba"):
+        pred_fun = estimator.predict_proba
+    else:
+        raise ValueError("The estimator has no predict or predict_proba method.")
+
+    # X is stacked n_grid times, and grid columns are replaced by replicated grid
+    n = X.shape[0]
+    n_grid = grid.shape[0]
+    X_eval = X.copy()
+
+    X_stacked = _safe_indexing(X_eval, np.tile(np.arange(n), n_grid), axis=0)
+    grid_stacked = _safe_indexing(grid, np.repeat(np.arange(n_grid), n), axis=0)
+    _safe_assign(X_stacked, values=grid_stacked, column_indexer=feature_indices)
+
+    # Predict on stacked data. Pick positive class probs for binary classification
+    preds = pred_fun(X_stacked)
+    if is_classifier(estimator) and preds.shape[1] == 2:
+        preds = preds[:, 1]
+
+    # Partial dependences are averages per grid block
+    pd_values = [
+        np.average(Z, axis=0, weights=sample_weight) for Z in np.split(preds, n_grid)
+    ]
+
+    return np.array(pd_values)
+
+
+def _calculate_pd_over_data(estimator, X, feature_indices, sample_weight=None):
+    """Calculates centered partial dependence over the data distribution.
+
+    It returns a numpy array of size (n, ) or (n, output_dimension).
+    """
+
+    # Select grid columns and remove duplicates (will compensate below)
+    grid = _safe_indexing(X, feature_indices, axis=1)
+
+    # np.unique() fails for mixed type and sparse objects
+    try:
+        ax = 0 if grid.shape[1] > 1 else None  # np.unique works better in 1 dim
+        _, ix, ix_reconstruct = np.unique(
+            grid, return_index=True, return_inverse=True, axis=ax
+        )
+        grid = _safe_indexing(grid, ix, axis=0)
+        compressed_grid = True
+    except (TypeError, np.AxisError):
+        compressed_grid = False
+
+    pd_values = _calculate_pd_brute_fast(
+        estimator,
+        X=X,
+        feature_indices=feature_indices,
+        grid=grid,
+        sample_weight=sample_weight,
+    )
+
+    if compressed_grid:
+        pd_values = pd_values[ix_reconstruct]
+
+    # H-statistics are based on *centered* partial dependences
+    column_means = np.average(pd_values, axis=0, weights=sample_weight)
+
+    return pd_values - column_means
+
+
+def h_statistic(
```

Comment:
```suggestion
@validate_params(
    {
        "estimator": [
            HasMethods(["fit", "predict"]),
            HasMethods(["fit", "predict_proba"]),
            HasMethods(["fit", "decision_function"]),
        ],
        "X": ["array-like", "sparse matrix"],
        "sample_weight": ["array-like", None],
        "features": ["array-like", Integral, str],
        "n_max": TODO: continue
    },
    prefer_skip_nested_validation=True,
)
def h_statistic(
```

Reply:
Polars and Pandas count as "array-like", right? Where could I find more docu on this?